sw/host/ot_certs/src/x509/tests.rs

```diff
@@ -0,0 +1,571 @@
+        &tmpl.certificate.subject_public_key_info;
+    let Value::Variable(Variable { name: name_x, .. }) = &public_key.x else {
+        panic!("the public key x must be a variable");
+    };
+    let Value::Variable(Variable { name: name_y, .. }) = &public_key.y else {
+        panic!("the public key y must be a variable");
+    };
+
+    let group = ecgroup_from_curve(curve);
+    let privkey = EcKey::<Private>::generate(&group).unwrap();
+    let mut ctx = BigNumContext::new().unwrap();
+    let mut x = BigNum::new().unwrap();
+    let mut y = BigNum::new().unwrap();
+    privkey
+        .public_key()
+        .affine_coordinates(&group, &mut x, &mut y, &mut ctx)
+        .unwrap();
+    const EC_X_Y_SIZE: usize = 32;
+    let x = x.to_vec_padded(EC_X_Y_SIZE.try_into().unwrap()).unwrap();
+    let y = y.to_vec_padded(EC_X_Y_SIZE.try_into().unwrap()).unwrap();
+
+    for var in &cert_with_offsets.variables {
+        // Make sure that every variable appear exactly once.
+        assert_eq!(
+            var.offsets.len(),
+            1,
+            "variable {} expected to appear once",
+            var.name
+        );
+        // Make sure that there is no conversion.
+        let off = &var.offsets[0];
+        assert_eq!(
+            var.source_type, off.target_type,
+            "variable {} expected to have the same source and target type",
+            var.name
+        );
+        // Generate a random value, except for the public key.
+        let bytes = if var.name == *name_x {
+            // Convert x to the DER representation, potentially adding some padding if necessary.
+            integers.insert(var.name.clone(), BigUint::from_bytes_be(&x));
+            x.clone()
+        } else if var.name == *name_y {
+            // Same.
+            integers.insert(var.name.clone(), BigUint::from_bytes_be(&y));
+            y.clone()
+        } else {
+            match off.target_type {
+                VariableType::ByteArray { size } => {
+                    let (bytes, value) = Vec::<u8>::random_value(size);
+                    byte_arrays.insert(var.name.clone(), value);
+                    bytes
+                }
+                VariableType::String { size } => {
+                    let (bytes, value) = String::random_value(size);
+                    strings.insert(var.name.clone(), value);
+                    bytes
+                }
+                VariableType::Integer { size } => {
+                    let (bytes, value) = BigUint::random_value(size);
+                    integers.insert(var.name.clone(), value);
+                    bytes
+                }
+            }
+        };
+        // Overwrite the certificate
+        assert_eq!(bytes.len(), off.target_type.size());
+        for (from, to) in zip(
+            &bytes,
+            &mut cert_with_offsets.cert[off.offset..off.offset + bytes.len()],
+        ) {
+            *to = *from;
+        }
```

Comment:
```suggestion
        cert_with_offsets.cert[off.offset..][..bytes.len()].copy_from_slice(&bytes);
```

Reply:
Thanks, I suspected that a much better way to write this ^^